### PR TITLE
[MISC] Update snap dependencies

### PR DESCRIPTION
**Pull request by dragomirp**
* Update PG to 14.11
* Add pysyncobj PPA

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -97,7 +99,8 @@ apps:
     environment:
       TCL_LIBRARY: $SNAP/usr/share/tcltk/tcl8.6
       "LD_LIBRARY_PATH": "$LD_LIBRARY_PATH:$SNAP/usr/lib/\
-      $CRAFT_ARCH_TRIPLET/blas:$SNAP/usr/lib/$CRAFT_ARCH_TRIPLET/lapack"
+      $CRAFT_ARCH_TRIPLET_BUILD_FOR/blas:$SNAP/usr/lib/\
```
Comment (dragomirp) [inline snap/snapcraft.yaml:102]: CRAFT_ARCH_TRIPLET is deprecated.

**Review by carlcsaposs-canonical — APPROVED** [review]

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # yamllint disable rule:line-length
 name: charmed-postgresql # you probably want to 'snapcraft register <name>'
 base: core22 # the base snap is the execution environment for this snap
-version: '14.10' # just for humans, typically '1.2+git' or '1.3.2'
+version: '14.11' # just for humans, typically '1.2+git' or '1.3.2'
```
Comment (taurus-forever) [inline snap/snapcraft.yaml:4]: Consider to delete those unrelated comments (here and above) the next ticket update.

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -240,7 +243,7 @@ parts:
       usr/share/doc/util-linux/copyright: licenses/COPYRIGHT-util-linux
       usr/share/doc/pgbouncer/copyright: licenses/COPYRIGHT-pgbouncer
       usr/share/doc/pgbackrest/copyright: licenses/COPYRIGHT-pgbackrest
-      usr/share/doc/prometheus-postgres-exporter/copyright: licences/COPYRIGHT-prometheus-postgres-exporter
+      usr/share/doc/prometheus-postgres-exporter/copyright: licenses/COPYRIGHT-prometheus-postgres-exporter
```
Comment (taurus-forever) [inline snap/snapcraft.yaml:246]: Eagle-eye!

**Review by taurus-forever — APPROVED** [review]
Nice!

**Pull request merged**